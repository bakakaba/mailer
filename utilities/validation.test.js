const validation = require("./validation");

test("missing required fields", () => {
  const result = validation.validateMail();
  expect(result).toMatchSnapshot();
});

test("invalid email", () => {
  const mail = {
    to: [{ email: "test+to@something" }],
    from: { email: "test+from@something" },
    subject: "hello world",
    body: "Test email"
  };
  const result = validation.validateMail(mail);
  expect(result).toMatchSnapshot();
});

test("valid mail", () => {
  const mail = {
    to: [{ email: "test+to@something.com" }],
    from: { email: "test+from@something.com" },
    subject: "hello world",
    body: "Test email"
  };
  const result = validation.validateMail(mail);
  expect(result.length).toBe(0);
});
