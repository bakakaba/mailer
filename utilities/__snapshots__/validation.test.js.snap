// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid email 1`] = `
Array [
  "test+from@something is not valid.",
  "test+to@something is not valid.",
]
`;

exports[`missing required fields 1`] = `
Array [
  "Mail is missing one or more required fields (to|from|subject|body).",
]
`;
